src/db/MariadbIfc.py

```diff
@@ -0,0 +1,504 @@
+        self.db_log.debug(f"Executing command: {cmd}")
+        cursor.execute(cmd)
+
+        for x in cursor:
+
+            self.db_log.debug(f"Adding result: {x}")
+            results.append((x[1],x[0]))
+
+        self.db_log.debug(f"Got results: {results}")
+
+        return results
+
+    def ResetDailyRoll(self):
+        """Resets the 'daily' boolean for all user profiles, allowing them to
+           perform another round of daily actions.
+
+            Input: N/A
+
+            Output: N/A.
+        """
+        cursor     = self.con.cursor(buffered=False)
+        cmd        = ""
+        result     = None
+
+        self.db_log.warning(f"Preparing to reset daily rolls.")
+        cmd = (self.user_cmds['reset_daily'])
+        self.db_log.debug(f"Executing daily roll reset command: {cmd}")
+
+        try:
+
+            cursor.execute(cmd)
+
+        except Exception as err:
+
+            self.db_log.error(f"Failed to reset daily value!: {err=}")
+
+    def SaveRoll(self,
+                 id      : Optional[str] = "x'fffffffffffffffffffffffffffffffe'",
+                 img     : Optional[str] = None,
+                 info    : dict          = None,
+                 profile : str           = None):
+        """Created a UUID for the given profile.  Assumes daily limits have
+           already been verified before calling.
+
+            Input: self - Pointer to the current object instance.
+                   id - user ID to link the profile to.
+                   info - the picture metadata to store.
+                   profile - The profile to link the image to.
+
+            Output: N/A.
+        """
+        cursor     = self.con.cursor(buffered=False)
+        cmd        = ""
+        entry      = profile#pic.loads(profile)
+        entry.info = info
+        owned      = None
+        result     = None
+
+
+        cursor.execute((self.prof_cmds['get_profile']) % id)
+        #The cursor object doesn't appear to actually provide a better way
+        #to determine if the cursor has a result.
+        for x in cursor:
+
+            result = x
+            self.db_log.debug(f"Cursor Response in loop: {x}")
+
+        if (result != None):
+
+            #Is this is a test command?
+            if (x[0] == self.db_cmds['default_id']):
+
+                self.db_log.warning(f"Rolled the test profile: {x}")
+
+            #Somehow the the roll is linked to an existing, non-test profile?
+            else:
+
+                #TODO: post an error message to the user to try again.
+                self.db_log.warning(f"Found an existing non-test profile: {x}  For roll {info}")
+
+        else:
+
+            cmd = (self.prof_cmds['put_new']) % (self.db_cmds['default_id'], id, entry.creator, entry.stats.agility, entry.stats.defense, entry.stats.endurance, entry.stats.luck, entry.stats.strength, entry.desc, entry.favorite, json.dumps(entry.info), entry.name, entry.rarity.value)
```

Comment:
dumping

Reply:
#dumps?

this was the cleanest way I had to both log and actually execute the command for each query.  The image info is stored as a JSON entry in the DB (since that's how the SD server returns it).